### PR TITLE
feat: add PUT /k/v1/preview/app/notifications/reminder.json

### PR DESCRIPTION
**Pull request by shisama**
New REST API `PUT /k/v1/preview/app/notifications/reminder.json` has already been added.
https://developer.cybozu.io/hc/ja/articles/900003822626

This adds a new method into KintoneRestAPIClient for the new REST API.

## What

This adds `updateReminderlNotifications()` method into `client.app` object.
The following is a TypeScript sample code for the method.

```typescript
import { KintoneRestAPIClient } from "@kintone/rest-api-client";
const updateNotifications = async (baseUrl: string, auth: Auth, app: App, notifications: Notifications) => {
  const client = new KintoneRestAPIClient({ baseUrl, auth });
  await this.client.app.updateReminderlNotifications({
    app,
    notifications,
  });
}
```

## How to test

This includes tests for this implementation. So just run the below.
`yarn test`

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [x] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

File: packages/rest-api-client/src/client/types/app/notification.ts [hunk 1]
```diff
@@ -50,6 +50,26 @@ export type PerRecordNotificationForResponse = {
...
+      }
+    | {
+        code: string;
+        daysLater: string | number;
+        time: string;
+      };
+  filterCond?: string;
+  title?: string;
+  targets: Array<{
+    entity: Entity;
```
Comment (shintaroNagata) [inline packages/rest-api-client/src/client/types/app/notification.ts:68]: With this definition, we can't specify `FIELD_ENTITY` to `type` property of `entity`.
Please modify the definition to enable to specify it.

File: packages/rest-api-client/docs/app.md [hunk 2]
```diff
@@ -1166,3 +1166,38 @@ Gets the Reminder Notification settings of the App.
 #### Reference
 
 - https://developer.kintone.io/hc/en-us/articles/900003958666
+
+### updateReminderNotifications
```
Comment (shintaroNagata) [inline packages/rest-api-client/docs/app.md:1170]: Please update the "table of contents" of this page.

File: packages/rest-api-client/src/client/__tests__/AppClient.test.ts [hunk 3]
```diff
@@ -957,6 +957,66 @@ describe("AppClient", () => {
     });
   });
 
+  describe("updateReminderlNotifications", () => {
```
Comment (shintaroNagata) [inline packages/rest-api-client/src/client/__tests__/AppClient.test.ts:960]: typo for `Reminder`
```suggestion
  describe("updateReminderNotifications", () => {
```

File: examples/rest-api-client-demo/src/app.ts [hunk 4]
```diff
@@ -593,4 +593,36 @@ export class App {
       console.log(error);
     }
   }
+
+  public async updateReminderlNotifications() {
```
Comment (shintaroNagata) [inline examples/rest-api-client-demo/src/app.ts:597]: typo for `Reminder`
```suggestion
  public async updateReminderNotifications() {
```

**Review by shintaroNagata — COMMENTED** [review]
@shisama 
Thank you for your PR! I left some review comments, so please check them.

**Comment by shisama** [comment]
@shintaroNagata Thank you for your comments. I've fixed them, so please check them again.

**Review by shintaroNagata — APPROVED** [review]
LGTM 👍

**Pull request merged**